File: DataFeedViews/OutputOptionsDataFeedView.vb
'*********************************************************************************************
' ST-Sim: A SyncroSim Module for the ST-Sim State-and-Transition Model.
'
' Copyright © 2007-2017 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.
'
'*********************************************************************************************

Imports System.Reflection
Imports System.Globalization
Imports SyncroSim.Core

<ObfuscationAttribute(Exclude:=True, ApplyToMembers:=False)>
Class OutputOptionsDataFeedView

    Const DEFAULT_TIMESTEP_VALUE As String = "1"

    Public Overrides Sub LoadDataFeed(dataFeed As DataFeed)

        MyBase.LoadDataFeed(dataFeed)

        Me.SetCheckBoxBinding(Me.CheckBoxSummarySC, DATASHEET_OO_SUMMARY_OUTPUT_SC_COLUMN_NAME)
        Me.SetTextBoxBinding(Me.TextBoxSummarySCTimesteps, DATASHEET_OO_SUMMARY_OUTPUT_SC_TIMESTEPS_COLUMN_NAME)
        Me.SetCheckBoxBinding(Me.CheckBoxSummarySCZeroValues, DATASHEET_OO_SUMMARY_OUTPUT_SC_ZERO_VALUES_COLUMN_NAME)
        Me.SetCheckBoxBinding(Me.CheckBoxSummaryTR, DATASHEET_OO_SUMMARY_OUTPUT_TR_COLUMN_NAME)
        Me.SetTextBoxBinding(Me.TextBoxSummaryTRTimesteps, DATASHEET_OO_SUMMARY_OUTPUT_TR_TIMESTEPS_COLUMN_NAME)
        Me.SetCheckBoxBinding(Me.CheckBoxSummaryTRCalcIntervalMean, DATASHEET_OO_SUMMARY_OUTPUT_TR_INTERVAL_MEAN_COLUMN_NAME)
        Me.SetCheckBoxBinding(Me.CheckBoxSummaryTRSC, DATASHEET_OO_SUMMARY_OUTPUT_TRSC_COLUMN_NAME)
        Me.SetTextBoxBinding(Me.TextBoxSummaryTRSCTimesteps, DATASHEET_OO_SUMMARY_OUTPUT_TRSC_TIMESTEPS_COLUMN_NAME)
        Me.SetCheckBoxBinding(Me.CheckBoxSummarySA, DATASHEET_OO_SUMMARY_OUTPUT_SA_COLUMN_NAME)
        Me.SetTextBoxBinding(Me.TextBoxSummarySATimesteps, DATASHEET_OO_SUMMARY_OUTPUT_SA_TIMESTEPS_COLUMN_NAME)
        Me.SetCheckBoxBinding(Me.CheckBoxSummaryTA, DATASHEET_OO_SUMMARY_OUTPUT_TA_COLUMN_NAME)
        Me.SetTextBoxBinding(Me.TextBoxSummaryTATimesteps, DATASHEET_OO_SUMMARY_OUTPUT_TA_TIMESTEPS_COLUMN_NAME)
        Me.SetCheckBoxBinding(Me.CheckBoxRasterSC, DATASHEET_OO_RASTER_OUTPUT_SC_COLUMN_NAME)
        Me.SetTextBoxBinding(Me.TextBoxRasterSCTimesteps, DATASHEET_OO_RASTER_OUTPUT_SC_TIMESTEPS_COLUMN_NAME)
        Me.SetCheckBoxBinding(Me.CheckBoxRasterTR, DATASHEET_OO_RASTER_OUTPUT_TR_COLUMN_NAME)
        Me.SetTextBoxBinding(Me.TextBoxRasterTRTimesteps, DATASHEET_OO_RASTER_OUTPUT_TR_TIMESTEPS_COLUMN_NAME)
        Me.SetCheckBoxBinding(Me.CheckBoxRasterAge, DATASHEET_OO_RASTER_OUTPUT_AGE_COLUMN_NAME)
        Me.SetTextBoxBinding(Me.TextBoxRasterAgeTimesteps, DATASHEET_OO_RASTER_OUTPUT_AGE_TIMESTEPS_COLUMN_NAME)
        Me.SetCheckBoxBinding(Me.CheckBoxRasterTST, DATASHEET_OO_RASTER_OUTPUT_TST_COLUMN_NAME)
        Me.SetTextBoxBinding(Me.TextBoxRasterTSTTimesteps, DATASHEET_OO_RASTER_OUTPUT_TST_TIMESTEPS_COLUMN_NAME)
        Me.SetCheckBoxBinding(Me.CheckBoxRasterST, DATASHEET_OO_RASTER_OUTPUT_ST_COLUMN_NAME)
        Me.SetTextBoxBinding(Me.TextBoxRasterSTTimesteps, DATASHEET_OO_RASTER_OUTPUT_ST_TIMESTEPS_COLUMN_NAME)
        Me.SetCheckBoxBinding(Me.CheckBoxRasterSA, DATASHEET_OO_RASTER_OUTPUT_SA_COLUMN_NAME)
        Me.SetTextBoxBinding(Me.TextBoxRasterSATimesteps, DATASHEET_OO_RASTER_OUTPUT_SA_TIMESTEPS_COLUMN_NAME)
        Me.SetCheckBoxBinding(Me.CheckBoxRasterTA, DATASHEET_OO_RASTER_OUTPUT_TA_COLUMN_NAME)
        Me.SetTextBoxBinding(Me.TextBoxRasterTATimesteps, DATASHEET_OO_RASTER_OUTPUT_TA_TIMESTEPS_COLUMN_NAME)
        Me.SetCheckBoxBinding(Me.CheckBoxRasterAATP, DATASHEET_OO_RASTER_OUTPUT_AATP_COLUMN_NAME)
        Me.SetTextBoxBinding(Me.TextBoxRasterAATPTimesteps, DATASHEET_OO_RASTER_OUTPUT_AATP_TIMESTEPS_COLUMN_NAME)

        Me.MonitorDataSheet(DATASHEET_TERMINOLOGY_NAME, AddressOf Me.OnTerminologyChanged, True)
        Me.AddStandardCommands()

    End Sub

    Public Overrides Sub EnableView(enable As Boolean)

        MyBase.EnableView(enable)

        If (enable) Then
            Me.EnableControls()
        End If

    End Sub

    Protected Overrides Sub OnRowsAdded(sender As Object, e As DataSheetRowEventArgs)

        MyBase.OnRowsAdded(sender, e)

        If (Me.ShouldEnableView()) Then
            Me.EnableControls()
        End If

    End Sub

    Protected Overrides Sub OnRowsDeleted(sender As Object, e As DataSheetRowEventArgs)

        MyBase.OnRowsDeleted(sender, e)

        If (Me.ShouldEnableView()) Then
            Me.EnableControls()
        End If

    End Sub

    Private Sub OnTerminologyChanged(ByVal e As DataSheetMonitorEventArgs)

        Dim t As String = CStr(e.GetValue("TimestepUnits", "Timestep")).ToLower(CultureInfo.InvariantCulture)

        Me.LabelSummarySCTimesteps.Text = t
        Me.LabelSummaryTRTimesteps.Text = t
        Me.LabelSummaryTRSCTimesteps.Text = t
        Me.LabelSummarySATimesteps.Text = t
        Me.LabelSummaryTATimesteps.Text = t
        Me.LabelRasterSCTimesteps.Text = t
        Me.LabelRasterTRTimesteps.Text = t
        Me.LabelRasterAgeTimesteps.Text = t
        Me.LabelRasterTSTTimesteps.Text = t
        Me.LabelRasterSTTimesteps.Text = t
        Me.LabelRasterSATimesteps.Text = t
        Me.LabelRasterTATimesteps.Text = t
        Me.LabelRasterAATPTimesteps.Text = t

    End Sub

    Private Sub EnableControls()

        'Text Boxes
        Me.TextBoxSummarySCTimesteps.Enabled = Me.CheckBoxSummarySC.Checked
        Me.TextBoxSummaryTRTimesteps.Enabled = Me.CheckBoxSummaryTR.Checked
        Me.TextBoxSummaryTRSCTimesteps.Enabled = Me.CheckBoxSummaryTRSC.Checked
        Me.TextBoxSummarySATimesteps.Enabled = Me.CheckBoxSummarySA.Checked
        Me.TextBoxSummaryTATimesteps.Enabled = Me.CheckBoxSummaryTA.Checked
        Me.TextBoxRasterSCTimesteps.Enabled = Me.CheckBoxRasterSC.Checked
        Me.TextBoxRasterTRTimesteps.Enabled = Me.CheckBoxRasterTR.Checked
        Me.TextBoxRasterAgeTimesteps.Enabled = Me.CheckBoxRasterAge.Checked
        Me.TextBoxRasterTSTTimesteps.Enabled = Me.CheckBoxRasterTST.Checked
        Me.TextBoxRasterSTTimesteps.Enabled = Me.CheckBoxRasterST.Checked
        Me.TextBoxRasterSATimesteps.Enabled = Me.CheckBoxRasterSA.Checked
        Me.TextBoxRasterTATimesteps.Enabled = Me.CheckBoxRasterTA.Checked
        Me.TextBoxRasterAATPTimesteps.Enabled = Me.CheckBoxRasterAATP.Checked

        'Timesteps labels
        Me.LabelSummarySCTimesteps.Enabled = Me.CheckBoxSummarySC.Checked
        Me.LabelSummaryTRTimesteps.Enabled = Me.CheckBoxSummaryTR.Checked
        Me.LabelSummaryTRSCTimesteps.Enabled = Me.CheckBoxSummaryTRSC.Checked
        Me.LabelSummarySATimesteps.Enabled = Me.CheckBoxSummarySA.Checked
        Me.LabelSummaryTATimesteps.Enabled = Me.CheckBoxSummaryTA.Checked
        Me.LabelRasterSCTimesteps.Enabled = Me.CheckBoxRasterSC.Checked
        Me.LabelRasterTRTimesteps.Enabled = Me.CheckBoxRasterTR.Checked
        Me.LabelRasterAgeTimesteps.Enabled = Me.CheckBoxRasterAge.Checked
        Me.LabelRasterTSTTimesteps.Enabled = Me.CheckBoxRasterTST.Checked
        Me.LabelRasterSTTimesteps.Enabled = Me.CheckBoxRasterST.Checked
        Me.LabelRasterSATimesteps.Enabled = Me.CheckBoxRasterSA.Checked
        Me.LabelRasterTATimesteps.Enabled = Me.CheckBoxRasterTA.Checked
        Me.LabelRasterAATPTimesteps.Enabled = Me.CheckBoxRasterAATP.Checked

        'Checkboxes
        Me.CheckBoxSummarySCZeroValues.Enabled = Me.CheckBoxSummarySC.Checked
        Me.CheckBoxSummaryTRCalcIntervalMean.Enabled = Me.CheckBoxSummaryTR.Checked

    End Sub

    Protected Overrides Sub OnBoundCheckBoxChanged(checkBox As System.Windows.Forms.CheckBox, columnName As String)

        MyBase.OnBoundCheckBoxChanged(checkBox, columnName)

        If (checkBox Is Me.CheckBoxSummarySC And Me.CheckBoxSummarySC.Checked And String.IsNullOrEmpty(Me.TextBoxSummarySCTimesteps.Text)) Then
            Me.SetTextBoxData(Me.TextBoxSummarySCTimesteps, DEFAULT_TIMESTEP_VALUE)
        ElseIf (checkBox Is Me.CheckBoxSummaryTR And Me.CheckBoxSummaryTR.Checked And String.IsNullOrEmpty(Me.TextBoxSummaryTRTimesteps.Text)) Then
            Me.SetTextBoxData(Me.TextBoxSummaryTRTimesteps, DEFAULT_TIMESTEP_VALUE)
        ElseIf (checkBox Is Me.CheckBoxSummaryTRSC And Me.CheckBoxSummaryTRSC.Checked And String.IsNullOrEmpty(Me.TextBoxSummaryTRSCTimesteps.Text)) Then
            Me.SetTextBoxData(Me.TextBoxSummaryTRSCTimesteps, DEFAULT_TIMESTEP_VALUE)
        ElseIf (checkBox Is Me.CheckBoxSummarySA And Me.CheckBoxSummarySA.Checked And String.IsNullOrEmpty(Me.TextBoxSummarySATimesteps.Text)) Then
            Me.SetTextBoxData(Me.TextBoxSummarySATimesteps, DEFAULT_TIMESTEP_VALUE)
        ElseIf (checkBox Is Me.CheckBoxSummaryTA And Me.CheckBoxSummaryTA.Checked And String.IsNullOrEmpty(Me.TextBoxSummaryTATimesteps.Text)) Then
            Me.SetTextBoxData(Me.TextBoxSummaryTATimesteps, DEFAULT_TIMESTEP_VALUE)
        ElseIf (checkBox Is Me.CheckBoxRasterSC And Me.CheckBoxRasterSC.Checked And String.IsNullOrEmpty(Me.TextBoxRasterSCTimesteps.Text)) Then
            Me.SetTextBoxData(Me.TextBoxRasterSCTimesteps, DEFAULT_TIMESTEP_VALUE)
        ElseIf (checkBox Is Me.CheckBoxRasterTR And Me.CheckBoxRasterTR.Checked And String.IsNullOrEmpty(Me.TextBoxRasterTRTimesteps.Text)) Then
            Me.SetTextBoxData(Me.TextBoxRasterTRTimesteps, DEFAULT_TIMESTEP_VALUE)
        ElseIf (checkBox Is Me.CheckBoxRasterAge And Me.CheckBoxRasterAge.Checked And String.IsNullOrEmpty(Me.TextBoxRasterAgeTimesteps.Text)) Then
            Me.SetTextBoxData(Me.TextBoxRasterAgeTimesteps, DEFAULT_TIMESTEP_VALUE)
        ElseIf (checkBox Is Me.CheckBoxRasterTST And Me.CheckBoxRasterTST.Checked And String.IsNullOrEmpty(Me.TextBoxRasterTSTTimesteps.Text)) Then
            Me.SetTextBoxData(Me.TextBoxRasterTSTTimesteps, DEFAULT_TIMESTEP_VALUE)
        ElseIf (checkBox Is Me.CheckBoxRasterST And Me.CheckBoxRasterST.Checked And String.IsNullOrEmpty(Me.TextBoxRasterSTTimesteps.Text)) Then
            Me.SetTextBoxData(Me.TextBoxRasterSTTimesteps, DEFAULT_TIMESTEP_VALUE)
        ElseIf (checkBox Is Me.CheckBoxRasterSA And Me.CheckBoxRasterSA.Checked And String.IsNullOrEmpty(Me.TextBoxRasterSATimesteps.Text)) Then
            Me.SetTextBoxData(Me.TextBoxRasterSATimesteps, DEFAULT_TIMESTEP_VALUE)
        ElseIf (checkBox Is Me.CheckBoxRasterTA And Me.CheckBoxRasterTA.Checked And String.IsNullOrEmpty(Me.TextBoxRasterTATimesteps.Text)) Then
            Me.SetTextBoxData(Me.TextBoxRasterTATimesteps, DEFAULT_TIMESTEP_VALUE)
        ElseIf (checkBox Is Me.CheckBoxRasterAATP And Me.CheckBoxRasterAATP.Checked And String.IsNullOrEmpty(Me.TextBoxRasterAATPTimesteps.Text)) Then
            Me.SetTextBoxData(Me.TextBoxRasterAATPTimesteps, DEFAULT_TIMESTEP_VALUE)
        End If

        Me.EnableControls()

    End Sub

End Class

File: DataFeedViews/OutputOptionsDataFeedView.cs
// A SyncroSim Package for developing state-and-transition simulation models using ST-Sim.
// Copyright © 2007-2018 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.

using System;
using System.Globalization;
using SyncroSim.Core;

namespace SyncroSim.STSim
{
    internal partial class OutputOptionsDataFeedView
    {
        public OutputOptionsDataFeedView()
        {
            InitializeComponent();
        }

        private const string DEFAULT_TIMESTEP_VALUE = "1";

        public override void LoadDataFeed(DataFeed dataFeed)
        {
            base.LoadDataFeed(dataFeed);

            this.SetCheckBoxBinding(this.CheckBoxSummarySC, Strings.DATASHEET_OO_SUMMARY_OUTPUT_SC_COLUMN_NAME);
            this.SetTextBoxBinding(this.TextBoxSummarySCTimesteps, Strings.DATASHEET_OO_SUMMARY_OUTPUT_SC_TIMESTEPS_COLUMN_NAME);
            this.SetCheckBoxBinding(this.CheckBoxSummarySCAges, Strings.DATASHEET_OO_SUMMARY_OUTPUT_SC_AGES_COLUMN_NAME);
            this.SetCheckBoxBinding(this.CheckBoxSummarySCZeroValues, Strings.DATASHEET_OO_SUMMARY_OUTPUT_SC_ZERO_VALUES_COLUMN_NAME);
            this.SetCheckBoxBinding(this.CheckBoxSummaryTR, Strings.DATASHEET_OO_SUMMARY_OUTPUT_TR_COLUMN_NAME);
            this.SetTextBoxBinding(this.TextBoxSummaryTRTimesteps, Strings.DATASHEET_OO_SUMMARY_OUTPUT_TR_TIMESTEPS_COLUMN_NAME);
            this.SetCheckBoxBinding(this.CheckBoxSummaryTRAges, Strings.DATASHEET_OO_SUMMARY_OUTPUT_TR_AGES_COLUMN_NAME);
            this.SetCheckBoxBinding(this.CheckBoxSummaryTRCalcIntervalMean, Strings.DATASHEET_OO_SUMMARY_OUTPUT_TR_INTERVAL_MEAN_COLUMN_NAME);
            this.SetCheckBoxBinding(this.CheckBoxSummaryTRSC, Strings.DATASHEET_OO_SUMMARY_OUTPUT_TRSC_COLUMN_NAME);
            this.SetTextBoxBinding(this.TextBoxSummaryTRSCTimesteps, Strings.DATASHEET_OO_SUMMARY_OUTPUT_TRSC_TIMESTEPS_COLUMN_NAME);
            this.SetCheckBoxBinding(this.CheckBoxSummarySA, Strings.DATASHEET_OO_SUMMARY_OUTPUT_SA_COLUMN_NAME);
            this.SetTextBoxBinding(this.TextBoxSummarySATimesteps, Strings.DATASHEET_OO_SUMMARY_OUTPUT_SA_TIMESTEPS_COLUMN_NAME);
            this.SetCheckBoxBinding(this.CheckBoxSummarySAAges, Strings.DATASHEET_OO_SUMMARY_OUTPUT_SA_AGES_COLUMN_NAME);
            this.SetCheckBoxBinding(this.CheckBoxSummaryTA, Strings.DATASHEET_OO_SUMMARY_OUTPUT_TA_COLUMN_NAME);
            this.SetTextBoxBinding(this.TextBoxSummaryTATimesteps, Strings.DATASHEET_OO_SUMMARY_OUTPUT_TA_TIMESTEPS_COLUMN_NAME);
            this.SetCheckBoxBinding(this.CheckBoxSummaryTAAges, Strings.DATASHEET_OO_SUMMARY_OUTPUT_TA_AGES_COLUMN_NAME);
            this.SetCheckBoxBinding(this.CheckBoxSummaryOmitSS, Strings.DATASHEET_OO_SUMMARY_OUTPUT_OMIT_SS_COLUMN_NAME);
            this.SetCheckBoxBinding(this.CheckBoxSummaryOmitTS, Strings.DATASHEET_OO_SUMMARY_OUTPUT_OMIT_TS_COLUMN_NAME);
            this.SetCheckBoxBinding(this.CheckBoxRasterSC, Strings.DATASHEET_OO_RASTER_OUTPUT_SC_COLUMN_NAME);
            this.SetTextBoxBinding(this.TextBoxRasterSCTimesteps, Strings.DATASHEET_OO_RASTER_OUTPUT_SC_TIMESTEPS_COLUMN_NAME);
            this.SetCheckBoxBinding(this.CheckBoxRasterTR, Strings.DATASHEET_OO_RASTER_OUTPUT_TR_COLUMN_NAME);
            this.SetTextBoxBinding(this.TextBoxRasterTRTimesteps, Strings.DATASHEET_OO_RASTER_OUTPUT_TR_TIMESTEPS_COLUMN_NAME);
            this.SetCheckBoxBinding(this.CheckBoxRasterAge, Strings.DATASHEET_OO_RASTER_OUTPUT_AGE_COLUMN_NAME);
            this.SetTextBoxBinding(this.TextBoxRasterAgeTimesteps, Strings.DATASHEET_OO_RASTER_OUTPUT_AGE_TIMESTEPS_COLUMN_NAME);
            this.SetCheckBoxBinding(this.CheckBoxRasterTST, Strings.DATASHEET_OO_RASTER_OUTPUT_TST_COLUMN_NAME);
            this.SetTextBoxBinding(this.TextBoxRasterTSTTimesteps, Strings.DATASHEET_OO_RASTER_OUTPUT_TST_TIMESTEPS_COLUMN_NAME);
            this.SetCheckBoxBinding(this.CheckBoxRasterST, Strings.DATASHEET_OO_RASTER_OUTPUT_ST_COLUMN_NAME);
            this.SetTextBoxBinding(this.TextBoxRasterSTTimesteps, Strings.DATASHEET_OO_RASTER_OUTPUT_ST_TIMESTEPS_COLUMN_NAME);
            this.SetCheckBoxBinding(this.CheckBoxRasterSA, Strings.DATASHEET_OO_RASTER_OUTPUT_SA_COLUMN_NAME);
            this.SetTextBoxBinding(this.TextBoxRasterSATimesteps, Strings.DATASHEET_OO_RASTER_OUTPUT_SA_TIMESTEPS_COLUMN_NAME);
            this.SetCheckBoxBinding(this.CheckBoxRasterTA, Strings.DATASHEET_OO_RASTER_OUTPUT_TA_COLUMN_NAME);
            this.SetTextBoxBinding(this.TextBoxRasterTATimesteps, Strings.DATASHEET_OO_RASTER_OUTPUT_TA_TIMESTEPS_COLUMN_NAME);
            this.SetCheckBoxBinding(this.CheckBoxRasterAATP, Strings.DATASHEET_OO_RASTER_OUTPUT_AATP_COLUMN_NAME);
            this.SetTextBoxBinding(this.TextBoxRasterAATPTimesteps, Strings.DATASHEET_OO_RASTER_OUTPUT_AATP_TIMESTEPS_COLUMN_NAME);

            this.MonitorDataSheet(Strings.DATASHEET_TERMINOLOGY_NAME, this.OnTerminologyChanged, true);
            this.AddStandardCommands();
        }

        public override void EnableView(bool enable)
        {
            base.EnableView(enable);

            if (enable)
            {
                this.EnableControls();
            }
        }

        protected override void OnRowsAdded(object sender, DataSheetRowEventArgs e)
        {
            base.OnRowsAdded(sender, e);

            if (this.ShouldEnableView())
            {
                this.EnableControls();
            }
        }

        protected override void OnRowsDeleted(object sender, DataSheetRowEventArgs e)
        {
            base.OnRowsDeleted(sender, e);

            if (this.ShouldEnableView())
            {
                this.EnableControls();
            }
        }

        private void OnTerminologyChanged(DataSheetMonitorEventArgs e)
        {
            string t = Convert.ToString(
                e.GetValue("TimestepUnits", "Timestep"), 
                CultureInfo.InvariantCulture).ToLower(CultureInfo.InvariantCulture);

            this.LabelSummarySCTimesteps.Text = t;
            this.LabelSummaryTRTimesteps.Text = t;
            this.LabelSummaryTRSCTimesteps.Text = t;
            this.LabelSummarySATimesteps.Text = t;
            this.LabelSummaryTATimesteps.Text = t;
            this.LabelRasterSCTimesteps.Text = t;
            this.LabelRasterTRTimesteps.Text = t;
            this.LabelRasterAgeTimesteps.Text = t;
            this.LabelRasterTSTTimesteps.Text = t;
            this.LabelRasterSTTimesteps.Text = t;
            this.LabelRasterSATimesteps.Text = t;
            this.LabelRasterTATimesteps.Text = t;
            this.LabelRasterAATPTimesteps.Text = t;
        }

        private void EnableControls()
        {
            //Text Boxes
            this.TextBoxSummarySCTimesteps.Enabled = this.CheckBoxSummarySC.Checked;
            this.TextBoxSummaryTRTimesteps.Enabled = this.CheckBoxSummaryTR.Checked;
            this.TextBoxSummaryTRSCTimesteps.Enabled = this.CheckBoxSummaryTRSC.Checked;
            this.TextBoxSummarySATimesteps.Enabled = this.CheckBoxSummarySA.Checked;
            this.TextBoxSummaryTATimesteps.Enabled = this.CheckBoxSummaryTA.Checked;
            this.TextBoxRasterSCTimesteps.Enabled = this.CheckBoxRasterSC.Checked;
            this.TextBoxRasterTRTimesteps.Enabled = this.CheckBoxRasterTR.Checked;
            this.TextBoxRasterAgeTimesteps.Enabled = this.CheckBoxRasterAge.Checked;
            this.TextBoxRasterTSTTimesteps.Enabled = this.CheckBoxRasterTST.Checked;
            this.TextBoxRasterSTTimesteps.Enabled = this.CheckBoxRasterST.Checked;
            this.TextBoxRasterSATimesteps.Enabled = this.CheckBoxRasterSA.Checked;
            this.TextBoxRasterTATimesteps.Enabled = this.CheckBoxRasterTA.Checked;
            this.TextBoxRasterAATPTimesteps.Enabled = this.CheckBoxRasterAATP.Checked;

            //Timesteps labels
            this.LabelSummarySCTimesteps.Enabled = this.CheckBoxSummarySC.Checked;
            this.LabelSummaryTRTimesteps.Enabled = this.CheckBoxSummaryTR.Checked;
            this.LabelSummaryTRSCTimesteps.Enabled = this.CheckBoxSummaryTRSC.Checked;
            this.LabelSummarySATimesteps.Enabled = this.CheckBoxSummarySA.Checked;
            this.LabelSummaryTATimesteps.Enabled = this.CheckBoxSummaryTA.Checked;
            this.LabelRasterSCTimesteps.Enabled = this.CheckBoxRasterSC.Checked;
            this.LabelRasterTRTimesteps.Enabled = this.CheckBoxRasterTR.Checked;
            this.LabelRasterAgeTimesteps.Enabled = this.CheckBoxRasterAge.Checked;
            this.LabelRasterTSTTimesteps.Enabled = this.CheckBoxRasterTST.Checked;
            this.LabelRasterSTTimesteps.Enabled = this.CheckBoxRasterST.Checked;
            this.LabelRasterSATimesteps.Enabled = this.CheckBoxRasterSA.Checked;
            this.LabelRasterTATimesteps.Enabled = this.CheckBoxRasterTA.Checked;
            this.LabelRasterAATPTimesteps.Enabled = this.CheckBoxRasterAATP.Checked;

            //Checkboxes
            this.CheckBoxSummarySCZeroValues.Enabled = this.CheckBoxSummarySC.Checked;
            this.CheckBoxSummarySCAges.Enabled = this.CheckBoxSummarySC.Checked;
            this.CheckBoxSummaryTRCalcIntervalMean.Enabled = this.CheckBoxSummaryTR.Checked;
            this.CheckBoxSummaryTRAges.Enabled = this.CheckBoxSummaryTR.Checked;
            this.CheckBoxSummarySAAges.Enabled = this.CheckBoxSummarySA.Checked;
            this.CheckBoxSummaryTAAges.Enabled = this.CheckBoxSummaryTA.Checked;
        }

        protected override void OnBoundCheckBoxChanged(System.Windows.Forms.CheckBox checkBox, string columnName)
        {
            base.OnBoundCheckBoxChanged(checkBox, columnName);

            if (checkBox == this.CheckBoxSummarySC && this.CheckBoxSummarySC.Checked & string.IsNullOrEmpty(this.TextBoxSummarySCTimesteps.Text))
            {
                this.SetTextBoxData(this.TextBoxSummarySCTimesteps, DEFAULT_TIMESTEP_VALUE);
            }
            else if (checkBox == this.CheckBoxSummaryTR && this.CheckBoxSummaryTR.Checked & string.IsNullOrEmpty(this.TextBoxSummaryTRTimesteps.Text))
            {
                this.SetTextBoxData(this.TextBoxSummaryTRTimesteps, DEFAULT_TIMESTEP_VALUE);
            }
            else if (checkBox == this.CheckBoxSummaryTRSC && this.CheckBoxSummaryTRSC.Checked & string.IsNullOrEmpty(this.TextBoxSummaryTRSCTimesteps.Text))
            {
                this.SetTextBoxData(this.TextBoxSummaryTRSCTimesteps, DEFAULT_TIMESTEP_VALUE);
            }
            else if (checkBox == this.CheckBoxSummarySA && this.CheckBoxSummarySA.Checked & string.IsNullOrEmpty(this.TextBoxSummarySATimesteps.Text))
            {
                this.SetTextBoxData(this.TextBoxSummarySATimesteps, DEFAULT_TIMESTEP_VALUE);
            }
            else if (checkBox == this.CheckBoxSummaryTA && this.CheckBoxSummaryTA.Checked & string.IsNullOrEmpty(this.TextBoxSummaryTATimesteps.Text))
            {
                this.SetTextBoxData(this.TextBoxSummaryTATimesteps, DEFAULT_TIMESTEP_VALUE);
            }
            else if (checkBox == this.CheckBoxRasterSC && this.CheckBoxRasterSC.Checked & string.IsNullOrEmpty(this.TextBoxRasterSCTimesteps.Text))
            {
                this.SetTextBoxData(this.TextBoxRasterSCTimesteps, DEFAULT_TIMESTEP_VALUE);
            }
            else if (checkBox == this.CheckBoxRasterTR && this.CheckBoxRasterTR.Checked & string.IsNullOrEmpty(this.TextBoxRasterTRTimesteps.Text))
            {
                this.SetTextBoxData(this.TextBoxRasterTRTimesteps, DEFAULT_TIMESTEP_VALUE);
            }
            else if (checkBox == this.CheckBoxRasterAge && this.CheckBoxRasterAge.Checked & string.IsNullOrEmpty(this.TextBoxRasterAgeTimesteps.Text))
            {
                this.SetTextBoxData(this.TextBoxRasterAgeTimesteps, DEFAULT_TIMESTEP_VALUE);
            }
            else if (checkBox == this.CheckBoxRasterTST && this.CheckBoxRasterTST.Checked & string.IsNullOrEmpty(this.TextBoxRasterTSTTimesteps.Text))
            {
                this.SetTextBoxData(this.TextBoxRasterTSTTimesteps, DEFAULT_TIMESTEP_VALUE);
            }
            else if (checkBox == this.CheckBoxRasterST && this.CheckBoxRasterST.Checked & string.IsNullOrEmpty(this.TextBoxRasterSTTimesteps.Text))
            {
                this.SetTextBoxData(this.TextBoxRasterSTTimesteps, DEFAULT_TIMESTEP_VALUE);
            }
            else if (checkBox == this.CheckBoxRasterSA && this.CheckBoxRasterSA.Checked & string.IsNullOrEmpty(this.TextBoxRasterSATimesteps.Text))
            {
                this.SetTextBoxData(this.TextBoxRasterSATimesteps, DEFAULT_TIMESTEP_VALUE);
            }
            else if (checkBox == this.CheckBoxRasterTA && this.CheckBoxRasterTA.Checked & string.IsNullOrEmpty(this.TextBoxRasterTATimesteps.Text))
            {
                this.SetTextBoxData(this.TextBoxRasterTATimesteps, DEFAULT_TIMESTEP_VALUE);
            }
            else if (checkBox == this.CheckBoxRasterAATP && this.CheckBoxRasterAATP.Checked & string.IsNullOrEmpty(this.TextBoxRasterAATPTimesteps.Text))
            {
                this.SetTextBoxData(this.TextBoxRasterAATPTimesteps, DEFAULT_TIMESTEP_VALUE);
            }

            this.EnableControls();
        }
    }
}
